oiva-core/src/main/java/fi/minedu/oiva/backend/core/security/CustomSuccessHandler.java

```diff
@@ -15,11 +14,9 @@
     @Override
     public void onAuthenticationSuccess(final HttpServletRequest request, final HttpServletResponse response, final Authentication authentication) throws ServletException, IOException {
         final HttpSession session = request.getSession();
-        final String redirect = (String) session.getAttribute("redirect");
-        if (StringUtils.isNotBlank(redirect)) {
-            this.setDefaultTargetUrl((String) session.getAttribute("redirect"));
-            session.removeAttribute("redirect");
-        }
+        session.removeAttribute("redirect");
```

Comment:
Tässä voisi käyttää `this.setAlwaysUseDefaultTargetUrl(true);`, jotta saataisiin säilytettyä vanha toiminnallisuus. Eli `defaultTargetUrl` muuttujaa käytetään destination urlina ja mahdollinen `DefaultSavedRequest` tyhjennetään sessiosta. Eli tuon pitäisi totella aina vain tuota asetettua `defaultTargetUrlia`.

Reply:
Selvitin asiaa hieman pidemmän kaavan kautta kuin alunperin.
Frontista on pakollista antaa redirect-url-parametri login-rajapintaa kutsuttaessa. Tämä parametri asetetaan palvelimen sessioon ja sen pitäisi olla siellä olemassa joka kerta kun sucesshandleria kutsutaan.

Uusien muutosten jälkeen alkuperäinen toiminnallisuus pitäisi olla ennallaan. Tuo `this.setAlwaysUseDefaultTargetUrl(true);` tosiaan aiheuttaa sen että ohjauksessa käytetään defaultTargetUrlia eikä mitään muuta tietoa.